pandas/tseries/tools.py

```diff
@@ -300,17 +300,17 @@ def _convert_listlike(arg, box, format):
         if infer_datetime_format and format is None:
             format = _guess_datetime_format_for_array(arg, dayfirst=dayfirst)
```

Comment:
I think the iso format code needs to be outside the else - it should apply whether or not the format was inferred or specified?